src/details/ArborX_DetailsBoundingVolumeHierarchyImpl.hpp

```diff
@@ -635,6 +360,239 @@ BoundingVolumeHierarchyImpl<DeviceType>::queryDispatch(
   Kokkos::Profiling::popRegion();
 }
 
+template <typename BVH, typename ExecutionSpace, typename Predicates,
+          typename Indices, typename Offset>
+inline void
+queryDispatch(SpatialPredicateTag, BVH const &bvh, ExecutionSpace const &space,
+              Predicates const &predicates, Indices &indices, Offset &offset,
+              Experimental::TraversalPolicy const &policy =
+                  Experimental::TraversalPolicy())
+{
+  queryDispatch(SpatialPredicateTag{}, bvh, space, predicates,
+                CallbackDefaultSpatialPredicate{}, indices, offset, policy);
+}
+
+template <typename BVH, typename ExecutionSpace, typename Predicates,
+          typename OutputView, typename OffsetView, typename Callback>
+inline std::enable_if_t<
+    std::is_same<typename Callback::tag, PostCallbackTag>::value>
+queryDispatch(SpatialPredicateTag, BVH const &bvh, ExecutionSpace const &space,
+              Predicates const &predicates, Callback const &callback,
+              OutputView &out, OffsetView &offset,
+              Experimental::TraversalPolicy const &policy =
+                  Experimental::TraversalPolicy())
+{
+  Kokkos::View<int *, ExecutionSpace> indices("indices", 0);
```

Comment:
This is the only place in the new code where the type for indices is hardcoded. I'm not sure if it's a bug or a feature. If we never specify `int *` anywhere in the code, would it mean that we can play around with different types, like using `short` for indices and offsets for smaller sized problems?

Reply:
You beat me to it.  I was going to say in principle we don't need to impose the integer type.

Reply:
This is just fine